app/controllers/items_controller.rb

```diff
@@ -2,18 +2,86 @@ class ItemsController < OauthController
   before_action :login
 
   def refile
+    @refile_request = RefileRequest.new
+    # TODO: Finish this in subsequent branch.
+    if params[:page] && params[:per_page]
+      @offset = ( params[:page].to_i - 1 ) * params[:per_page].to_i
+    else
+      @offset = 0
+    end
+
+    date_start = params[:date_start].present? ? params[:date_start] : nil
+    date_end = params[:date_end].present? ? params[:date_end] : nil
+    @refiles = get_refiles(date_start,date_end,params[:page],params[:per_page])
   end
   
   def update_metadata
+    @barcodes = params[:notice].present? && params[:notice]["barcodes"].present? ? params[:notice]["barcodes"] : []
+    @protect_cgd = params[:notice].present? && params[:notice]["protect_cgd"].present? ? params[:notice]["protect_cgd"] : false
   end
 
   def send_metadata
-    barcodes = params[:barcodes]
-    puts barcodes
-    flash[:notice] = "Metadata updated for #{barcodes}"
-    redirect_to request.referrer
+    # don't let it fool you. params[:barcodes] is text.
+    barcodes = params[:barcodes].strip.split(/[\W\s]+/)
+    message = Message.new(barcodes: barcodes, protect_cgd: params[:protect_cgd], action: 'update')
+    if message.valid?
+      message.send_update_message_to_sqs
+      invalid_barcodes = []
+    elsif message.valid_barcodes.present? 
+      invalid_barcodes = message.invalid_barcodes
+      message.barcodes = message.valid_barcodes
```

Comment:
We talked about `send_update_message_to_sqs` owning the logic of only sending valid/well-formed barcodes through instead of the whole `barcodes` Array.